### DISCUSSION TITLE
Effective heat transfer coefficient for evaporator/condenser: Valid for transient behaviour?

### DISCUSSION THREAD

**Discussion by RouvenZ** [lbl-srg/modelica-buildings#3409] [Q&A] [answered]
Hello all,

within the model of the [evaporator/condenser](Buildings/Fluid/HeatExchangers/EvaporatorCondenser.mo), at line [41](https://github.com/lbl-srg/modelica-buildings/blob/27caecbdb5c698443699cc3e41a2ed37d8a49847/Buildings/Fluid/HeatExchangers/EvaporatorCondenser.mo#L41), $UA_{eff}$ is defined as $UA_{eff}=\frac{\varepsilon \cdot c_p \cdot \dot{m} }{1-\varepsilon}$  in agreement with the [paper provided by the authors](https://2017.international.conference.modelica.org/proceedings/html/submissions/ecp17132219_CimminoWetter.pdf). A simple derivation for this would come from the definition of $\varepsilon=\frac{T_{in}-T_{out}}{T_{in}-T_{ref}}$, such that $\dot{Q} = \dot{m} \cdot c_p \cdot \varepsilon \cdot (T_{ref}-T_{in}) = \dot{m} \cdot c_p \cdot \frac{\varepsilon}{1 - \varepsilon} \cdot (T_{ref}-T_{out})$. 
However, this definition of $\varepsilon$- to my knowledge - only holds for the steady state case. Nonetheless, $UA_{eff}$ is also used if other energy dynamics are chosen. Also, the definition of $\varepsilon:=\frac{T_{in}-T_{out}}{T_{in}-T_{ref}}$ is not used in the set of equations within the model as far as I can tell. If one would use it in case of a heat pump model, I assume it to overdetermine the model by forcing $\frac{\partial U}{\partial t}=0$. 

If one runs a test (e.g., [ScrollWaterToWater_Dynamic](Buildings/Fluid/HeatPumps/Validation/ScrollWaterToWater_Dynamic.mo) and takes the specific numbers for the evaporator (exemplary) $T_{in}$ (i.e., `sou1.medium.T`) , $T_{out} = T_{volume}$ (i.e., `heaPum1.eva.vol.T` and $T_{ref}$ (i.e., `heaPum1.com.port_a.T`) comparing them to the effectiveness (`heaPum1.eva.eps`), these values differ for transient behaviour but are equal for the steady state case (as one would expect from $\frac{\partial U}{\partial t}=0$).

My question is therefore the following: How would one argue utilizing $\varepsilon := \frac{T_{in} - T_{out}}{T_{in} - T_{ref}}$ as a step to ${UA}_{eff} := \frac{\varepsilon \cdot c_p \cdot \dot {m} } {1 - \varepsilon}$, expecting the latter to also be valid for transient behaviour while the former clearly isn't?

I am looking forward to your reply on this - have a nice day y'all!

Cheers
Rouven

**Comment by RouvenZ** [comment]
Hello all,

to add to this, I added the parameters 
`Real eff_eva_control = (sou1.medium.T - heaPum1.eva.vol.T)/(sou1.medium.T - heaPum1.com.port_a.T);`
`Real eff_con_control = (loa1.medium.T - heaPum1.con.vol.T)/(loa1.medium.T - heaPum1.com.port_b.T);`
without further changes to the [model ](Buildings/Fluid/HeatPumps/Validation/ScrollWaterToWater_Dynamic.mo) and plotted them against the values for the effectiveness as described above. As one can see, they are identical in steady state ($0 s\leq t \leq 300 s$ and $550 s\leq t \leq 800 s$) as expected but differ while not in steady state yet ($500 s\leq t \leq 550 s$). 

I am sure there are good arguments utilizing $\varepsilon := \frac{T_{in} - T_{out}}{T_{in} - T_{ref}}$ as a step towards arriving at ${UA}_{eff} := \frac{\varepsilon \cdot c_p \cdot \dot {m} } {1 - \varepsilon}$ even though the first one is not valid for transient behaviour. I sadly did not find that stated explicitly anywhere (hopefully I did not overlook it). 
@mwetter (please excuse dragging you into this conversation directly): When you published your very informative and helpful [paper on this in 2017](https://2017.international.conference.modelica.org/proceedings/html/submissions/ecp17132219_CimminoWetter.pdf), what has been your opinion on that, has this been of any concern at all? 

![230712 Heatpump Eva Control](https://github.com/lbl-srg/modelica-buildings/assets/75694408/7e447e22-8c08-4ba1-8b5e-2f27a2cf6b61)
![230712 Heatpump Con Control](https://github.com/lbl-srg/modelica-buildings/assets/75694408/b7c22e0a-ccfa-4f49-88f6-af00370894bc)

**Comment by mwetter** [comment, marked as answer]
@RouvenZ : The epsilon-NTU methodology is applicable for steady-state models. It is the solution of the differential equation for the heat transfer, integrated along the flow path under steady-state assumption. You are right that it is here also used for the dynamic simulation. This is an approximation, without it we would have to use a finite difference model for the heat exchanger which would be computationally much more demanding.
Please be aware that this heat pump model also has other assumptions that idealize its dynamic response: For example, it does not track the dynamic distribution of the refrigerant. Such models do exists in commercial Modelica libraries, but such models are computationally very demanding and compute too slow for practical use in annual whole building energy simulation.
